ethcore/wasm/src/env.rs

```diff
@@ -14,183 +14,270 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Wasm env module bindings
-
-use parity_wasm::elements::ValueType::*;
-use parity_wasm::interpreter::{self, UserFunctionDescriptor};
-use parity_wasm::interpreter::UserFunctionDescriptor::*;
-use super::runtime::{Runtime, UserTrap};
-
-pub const SIGNATURES: &'static [UserFunctionDescriptor] = &[
-	Static(
-		"_storage_read",
-		&[I32; 2],
-		None,
-	),
-	Static(
-		"_storage_write",
-		&[I32; 2],
-		None,
-	),
-	Static(
-		"_balance",
-		&[I32; 2],
-		None,
-	),
-	Static(
-		"_ext_malloc",
-		&[I32],
-		Some(I32),
-	),
-	Static(
-		"_ext_free",
-		&[I32],
+//! Env module glue for wasmi interpreter
+
+use std::cell::RefCell;
+use wasmi::{
+	self, Signature, Error, FuncRef, FuncInstance, MemoryDescriptor,
+	MemoryRef, MemoryInstance,
+};
+
+/// Internal ids all functions runtime supports. This is just a glue for wasmi interpreter
+/// that lacks high-level api and later will be factored out
+pub mod ids {
+	pub const STORAGE_WRITE_FUNC: usize = 0;
```

Comment:
I don't see the point of this 10-step numbering.
Also, maybe we can use some macro which will generate C-like enum, so we don't have to number this constants by hand?

Reply:
It's implementation detail, will refactored once more high-level api for wasmi is available

Reply:
Ok, I think I'm fine with it.